fix(target): disable target creation if connectUrl is left empty

tthvo opened the pull request:
Fixes #565 

**Fixes**

- Disable target creation if connectUrl is left empty in both cases (i.e. clicking Create button and pressing Enter).
- Target labels/annotations are changed to generic object since JSON is not mapped into this Map type and fields are added as general support for any objects. See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#setting_object_properties).

**Chores**

- Remove subscription to JVMDiscovery notification since `context.target.targets` will be called to return new array of targets.
- Refactor to use common local storage utilities.

andrewazores commented on src/app/TargetSelect/TargetSelect.tsx:
```diff
@@ -133,55 +109,45 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props)
 
+  const selectTargetFromCache = React.useCallback(
+    (targets) => {
+      if (targets.length) {
```
I think this outer if/else can actually be removed. If the `targets` array is empty then the `targets.some()` call later will trivially return `false` and the same handling case runs either way.

```bash
$ node
> ['a'].some(v => true)
true
> [].some(v => true)
false
```

tthvo replied:
Right! I will clean up this part too!

andrewazores commented:
Testing this out manually, it seems like something is causing a LOT of repeated API requests to the backend. I guess some component is getting re-rendered continuously and causing a request or series of requests each time. Just visiting the Recordings or Events views and selecting a target is enough to trigger it.

tthvo commented:
> Testing this out manually, it seems like something is causing a LOT of repeated API requests to the backend. I guess some component is getting re-rendered continuously and causing a request or series of requests each time. Just visiting the Recordings or Events views and selecting a target is enough to trigger it.

Ah yess I saw it now. Sorry didnt see it yesterday. I will have a look to see which commit is causing this.

tthvo commented:
Seemed like I ran into the classic indefinite effects due to ever-changing deps for effect hooks. Fixed it now :D but I am checking an odd warning when select another target from no target-selected state.

```log
Cannot update a component (`TargetView`) while rendering a different component (`TargetSelect`). To locate the bad setState() call inside `TargetSelect`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
```

tthvo commented:
It should be ready for reviews again now....

> Seemed like I ran into the classic indefinite effects due to ever-changing deps for effect hooks. Fixed it now :D but I am checking an odd warning when select another target from no target-selected state.
> 
> ```
> Cannot update a component (`TargetView`) while rendering a different component (`TargetSelect`). To locate the bad setState() call inside `TargetSelect`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
> ```

The issue for this is that i was trying to avoid depending on the current target state of `TargetSelect` as callback will change if state changes, so I used setState which returns the currently selected target. However, we are calling `context.target.setTarget()` from inside here, which triggers other components to render and React is pretty much against this.

So, I reverted back to use state and added a comment to warn about this. Perhaps after #429, we can resolve this....
